airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -962,6 +963,19 @@ public Optional<ActorCatalog> getActorCatalog(final UUID actorId,
 
   }
 
+  public Optional<ActorCatalogFetchEvent> getMostRecentActorCatalogFetchEventForSource(final UUID sourceId) throws IOException {
```

Comment:
I think that we should add a test in `ConfigRepositoryE2EReadWriteTest` for this method.

Reply:
We are still missing a test here. I have added one [here](https://github.com/airbytehq/airbyte/pull/18125) maybe you can branch out of this PR to add a test which use the mocks I have added in `MockData.java`